libopae/plugins/xfpga/properties.c

```diff
@@ -224,15 +224,15 @@ fpga_result __FPGA_API__ xfpga_fpgaUpdateProperties(fpga_token token,
 	// only set socket id if we have it on sysfs
 	result = sysfs_get_fme_path(_token->device_instance,
 			_token->subdev_instance, spath);
-	if (FPGA_OK != result)
-		return result;
+	if (FPGA_OK == result) {
 
-	resval = sysfs_parse_attribute64(spath,
-		FPGA_SYSFS_SOCKET_ID, &value);
+		resval = sysfs_parse_attribute64(spath,
+			FPGA_SYSFS_SOCKET_ID, &value);
 
-	if (0 == resval) {
-		_iprop.socket_id = (uint8_t)value;
-		SET_FIELD_VALID(&_iprop, FPGA_PROPERTY_SOCKETID);
+		if (0 == resval) {
+			_iprop.socket_id = (uint8_t)value;
+			SET_FIELD_VALID(&_iprop, FPGA_PROPERTY_SOCKETID);
+		}
 	}
```

Comment:
Should there be..
```c
else
   return result;
```

Reply:
else return result works in PF , But fails in VF , VF doesn't get socket  ID

Reply:
ah, good point.